auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -1160,22 +1352,95 @@ private void consumeContent(final XMLStreamReader reader) throws XMLStreamExcept
             }
         }
 
-        private boolean validNamespace(final String namespace) {
-            return ELYTRON_1_0.equals(namespace) || ELYTRON_1_0_1.equals(namespace);
+        private boolean validateDigitalSignature(String path, String name) throws IllegalStateException {
+            try {
+                DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+                dbf.setNamespaceAware(true);
+                Document doc = dbf.newDocumentBuilder().parse(path);
+                NodeList nl = doc.getElementsByTagNameNS(XMLSignature.XMLNS, "Signature");
+                if (nl.getLength() == 0) {
+                    throw new RealmUnavailableException("Cannot find Signature element");
+                }
+                XMLSignatureFactory fac = XMLSignatureFactory.getInstance("DOM");
+                DOMValidateContext valContext = new DOMValidateContext(publicKey, nl.item(0));
+                XMLSignature signature = fac.unmarshalXMLSignature(valContext);
+                boolean coreValidity = signature.validate(valContext);
+                ElytronMessages.log.tracef("FileSystemSecurityRealm - verification against signature for credential [%s] = %b", name, coreValidity);
+                return coreValidity;
+           } catch (ParserConfigurationException | IOException | MarshalException | XMLSignatureException | org.xml.sax.SAXException e) {
```

Comment:
Just to check, is there a reason for using `org.xml.sax.SAXException` explicitly?

Reply:
I missed that when adding the imports, importing it instead now